### ISSUE TITLE
Dask backend auto-scattering overloads scheduler memory

### ISSUE THREAD

**Issue by rikturr** [joblib/joblib#1154]
Cross-posting from https://github.com/dask/dask-ml/issues/789, but I think the better home for this might be here
--- 

I seem to be running into problems using the Dask distributed backend for joblib with scikit-learn classes. This notebook has the full reproducible example using a `FargateCluster`. The issue does not happen with a `LocalCluster`: https://nbviewer.jupyter.org/gist/rikturr/66427bd13e692726044b4903a790f013

this part fails with ~50MB data size:

```python
with joblib.parallel_backend('dask'):
        search.fit(data, target)
```

It also fails if I scatter the object before-hand:
```python
client.scatter([data, target])
with joblib.parallel_backend('dask'):
        search.fit(data, target)
```

It works properly if I manually scatter within `parallel_backend` like so:
```python
with joblib.parallel_backend('dask', scatter=[data, target]):
        search.fit(data, target)
```
This leads me to believe something is happening with the auto-scattering causing lots of memory to be passed through the scheduler at once

**Comment by ogrisel** [comment]
This might have been fixed in the master branch by #1133. Could you confirm?

**Comment by ogrisel** [comment]
Actually, I am not sure whether this is related to the problem @pierreglaser fixed in #1133 or not. I would not have thought the excessive scattering pressure fixed by #1133 would have caused the scheduler to get overloaded but rather the client to worker communication (or the worker to worker communication) to be overloaded.

With the over-scattering issue, the scheduler would still have to keep track of where the data are scattered but I would not expect the memory usage of the scheduler to blow-up like this.

**Comment by rikturr** [comment]
Just tested - master is working great! Tried it out with a 5GB object and the scheduler gets up to 5GB for a few seconds then back down once the workers take over. I was previously on 1.0 so something that happened in master since seems to have fixed it :)

**Comment by ogrisel** [comment]
The only change in master is the #1133. I would not have suspected that this would case the scheduler memory to grow like this but thanks very much for the report and the feedback.

We shall release a new version of joblib with this fix soon then. It's the first time we fix a bug in anticipation prior to actually experiencing it our-selves ;)

**Comment by ogrisel** [comment]
I released joblib 1.0.1 with this fix. Closing.
